### PR TITLE
Add dostrip for EAPI 7 (cherry-pick from portage-mgorny)

### PR DESCRIPTION
**Pull request by zmedico**
The series contains dostrip patches cherry picked from portage-mgorny:

commit https://github.com/mgorny/portage/commit/38273616460fef9b36d3862f366f49f8ba5e7911
Author: Michał Górny <mgorny@gentoo.org>
Date:   2018-03-25 14:37:44 +0200

    Add dostrip for EAPI 7
    
    Bug: https://bugs.gentoo.org/203891
    Closes: https://github.com/mgorny/portage/issues/8

commit https://github.com/mgorny/portage/commit/ddb250cf5291a83cd5d44910428f666445147448
Author: Michał Górny <mgorny@gentoo.org>
Date:   2018-03-25 14:32:40 +0200

    Allow individually allowing strip w/ RESTRICT=strip
    
    Update RESTRICT=strip to control the default strip inclusion list
    as specified in EAPI 7 rather than disabling stripping entirely.
    This makes it possible to strip individual files.

commit https://github.com/mgorny/portage/commit/ef5f2d3574d4f73d6e5fce7e3e798e2a05cc1893
Author: Michał Górny <mgorny@gentoo.org>
Date:   2018-03-25 14:31:36 +0200

    estrip: Report pre-stripped files even with RESTRICT=strip
    
    The purpose of RESTRICT=strip is to prevent files from being stripped,
    not to silence QA checks about pre-stripped files.

commit https://github.com/mgorny/portage/commit/46f0f9cf8d02ab498ab7b4fec4419e507433839b
Author: Michał Górny <mgorny@gentoo.org>
Date:   2018-03-25 14:11:20 +0200

    Introduce control variables for estrip

commit https://github.com/mgorny/portage/commit/5ccafc3662ada431e5c75ec35aeb06cc86ee53ba
Author: Michał Górny <mgorny@gentoo.org>
Date:   2018-03-25 14:05:19 +0200

    estrip: Use queue/dequeue logic

commit https://github.com/mgorny/portage/commit/ee10e26763051396eea077a640c47796674403cc
Author: Michał Górny <mgorny@gentoo.org>
Date:   2018-03-25 13:16:24 +0200

    Disarm prepstrip & prepallstrip
    
    Move stripping logic to a new estrip helper (that is not exposed
    to ebuilds) and make prepstrip & prepallstrip do nothing.

commit https://github.com/mgorny/portage/commit/ae52aceeb86253a8b5742b287229614340017e55
Author: Michał Górny <mgorny@gentoo.org>
Date:   2018-03-24 14:03:02 +0100

    prepstrip: Disable parallel work

Michał Górny (7):
  prepstrip: Disable parallel work
  Disarm prepstrip & prepallstrip
  estrip: Use queue/dequeue logic
  Introduce control variables for estrip
  estrip: Report pre-stripped files even with RESTRICT=strip
  Allow individually allowing strip w/ RESTRICT=strip
  Add dostrip for EAPI 7

 bin/eapi.sh                     |   4 +
 bin/ebuild-helpers/prepallstrip |  11 +-
 bin/ebuild-helpers/prepstrip    | 401 +-----------------------------------
 bin/ebuild.sh                   |   2 +-
 bin/estrip                      | 438 ++++++++++++++++++++++++++++++++++++++++
 bin/misc-functions.sh           |   6 +-
 bin/phase-helpers.sh            |  29 +++
 bin/save-ebuild-env.sh          |   4 +-
 8 files changed, 486 insertions(+), 409 deletions(-)
 create mode 100755 bin/estrip

### CODE REVIEW

File: bin/misc-functions.sh [hunk 1]
```diff
@@ -218,12 +218,16 @@ install_qa_check() {
 		)
 	done < <(printf "%s\0" "${qa_checks[@]}" | LC_ALL=C sort -u -z)
 
-	export STRIP_MASK
+	export STRIP
```
Comment (mgorny) [inline bin/misc-functions.sh]: I don't think I did that. Wouldn't this break some of the ebuilds relying on STRIP_MASK?

File: bin/ebuild-helpers/prepallstrip [hunk 2]
```diff
@@ -1,11 +1,8 @@
 #!/bin/bash
```
Comment (mgorny) [inline bin/ebuild-helpers/prepallstrip]: You may want to keep `prepallstrip` and `prepstrip` working. Some ebuilds are still using them (with `RESTRICT=strip`), and they will no longer be stripped in old EAPIs.

File: bin/misc-functions.sh (hunk 1 above)
Comment (zmedico) [reply bin/misc-functions.sh]: Rebase error, fixed now.

**Comment by zmedico** [comment]
Closing in favor of https://github.com/gentoo/portage/pull/281.

**Pull request closed without merging**